### PR TITLE
Bugfix/icons

### PR DESCRIPTION
**Pull request by Sadhill94**
TLDR;
- react and svg icon collisions is a bitch
- we don't import svg as React component same as before
- instead, when adding new icon you need to create a new IconNameOfSvg.tsx (have a look at others like IconCopy.ts or IconHydraBridgeLogo for an example)
- add it to the IconKeys on the Atoms/Icons/Icon.tsx file
- **render icons in others component didn't change, you still pass the same props and the name in camelCase**

It's a known issue on react, there is plenty on internet referenced with this behavior, importing simple svg doesn't seems to be an issue, **however**, when you have to deal with custom svg that have mask id or fill reference, it become a pain in the ass.

Due to perpetual conflicts with id collision, i had to find a way to avoid that. 
I went to integrate dynamic id at first, but some svg still use the mask(url#id) reference to fill the colors.In the end it didn't change much to have an id on the svg root element. Still buggy shit hapenned.
So i went for an approach to allow more customization of the svg itself, where the main icon component handler can receive and id as a props if you still need to target it, otherwise it will generate a random ID out of the box and pass it to the lazy imported icon component.

In that component, i will be able to change easily the mask(#id) reference with that id.
Eg for the hydrabridge logo:
```
<filter
  id={`${props.id}-1`}
  x="0"
  y="0"
  width="15.61rem"
  height="40rem"
  filterUnits="userSpaceOnUse"
>
  <feOffset dy="3" />
  <feGaussianBlur stdDeviation="3" result="blur" />
  <feFlood floodColor="#0080ff" floodOpacity="0.188" />
  <feComposite operator="in" in2="blur" />
</filter>
<filter
  id={`${props.id}-2`}
  x="0"
  y="0"
  width="15.61rem"
  height="40rem"
  filterUnits="userSpaceOnUse"
>
....
 <g
  transform="matrix(1, 0, 0, 1, 32.49, 66.33)"
  filter={`url(#${props.id}-1)`}
>
  <path
    id={`${props.id}-3`}
    data-name="Path 355"
    d="xxx"
    transform="translate(7.08 27)"
    fill="#fff"
  />
</g>
```
This way we don't have any collision between id and the svg render correctly !

Second challenge: the snapshot kept failling because of these dynamic id...
just added this line in the setupTest.ts and it solve the whole thing
```
jest.mock("./common/components/Atoms/Icons/Icon", () => "svg");
```
So instead of have my snapshot fail with this result
```
<svg
  className="sc-gsDKAQ csGOgr"
  color="#FFFFFF"
  fill="currentColor"
  height="2.4rem"
  id="5209.3159726668255"
  isopen="0"
  viewBox="0 0 20 20"
  width="2.4rem"
  xmlns="http://www.w3.org/2000/svg"
>
  <path d"xxx" />
</svg>
```
It generate this:
```
<svg
  className="sc-gsDKAQ csGOgr"
  height="2.4rem"
  name="cutArrowDown"
  style="margin-left: 1.7rem;"
  width="2.4rem"
/>
```
So the snapshots are not affected anymore with these dynamic id.

Have to say, have spent quite some time on this fuckery, however it was something to address because adding hacks or workaround when using icons would be a mess later on.

### CODE REVIEW

**Comment by Sadhill94** [comment]
@rimatik you said you have some comment on this one ?

**Review by rimatik — APPROVED** [review]
